src/utils.ts

```diff
@@ -0,0 +1,15 @@
+export async function stat(
+  path: string,
+  test: (info: Deno.FileInfo) => boolean,
+) {
+  try {
+    const info = await Deno.lstat(path);
+    return test(info);
```

Comment:
Maybe we should just return `Deno.FileInfo | undefined` from this function? Then we could do a stat call once in the code above. Right now we're doing possibly two stat calls per path instead of max 1.

Reply:
Actually I see your point, can you merge this as is (if there are no other issues), and I'll do the changes in the other pr

Reply:
Sure.